### ISSUE TITLE
Custom Template Variable "hide" value not preserved on import

### ISSUE THREAD

**Issue by octalthorpe** [grafana/grafana#5869]
* **I'm submitting a ...**
- [X] Bug report

- What Grafana version are you using?
Version 3.0.4 (commit: v3.0.4)

- What datasource are you using?
Influxdb

- What OS are you running grafana on?
Linux (ubuntu)

- What did you do?
Imported a dashboard that includes template variables, that were supposed to be hidden, value of "2"
I am using template variables to allow us to automate the generation of multiple dashboards per environments.

- What was the expected result?
I expect the dashboard that is imported to keep the hidden value for the template variable set in the imported source

- What happened instead?
Template variables were not hidden, reset to a value of "0", dashboard shows variables, not the end of the world but would like to have them hidden.

Template json before input
```
{
  "list": [
    {
      "allFormat": "glob",
      "current": {
        "text": "ion-staging",
        "value": "ion-staging"
      },
      "datasource": null,
      "hide": 2,
      "includeAll": false,
      "multi": false,
      "name": "deployment",
      "options": [
        {
          "selected": true,
          "text": "ion-staging",
          "value": "ion-staging"
        }
      ],
      "query": "ion-staging",
      "refresh": 0,
      "type": "custom"
    },
    {
      "allFormat": "glob",
      "current": {
        "text": "ion-staging-app",
        "value": "ion-staging-app"
      },
      "datasource": null,
      "hide": 2,
      "includeAll": false,
      "multi": false,
      "name": "app_elb_name",
      "options": [
        {
          "selected": true,
          "text": "ion-staging-app",
          "value": "ion-staging-app"
        }
      ],
      "query": "ion-staging-app",
      "refresh": 0,
      "type": "custom"
    },
    {
      "allFormat": "glob",
      "current": {
        "text": "staging",
        "value": "staging"
      },
      "datasource": null,
      "hide": 2,
      "includeAll": false,
      "multi": false,
      "name": "deployment_short",
      "options": [
        {
          "selected": true,
          "text": "staging",
          "value": "staging"
        }
      ],
      "query": "staging",
      "refresh": 0,
      "type": "custom"
    }
  ]
}
```
Template json after input
```
  "list": [
    {
      "allFormat": "glob",
      "current": {
        "text": "ion-staging",
        "value": "ion-staging"
      },
      "datasource": null,
      "hide": 0,
      "includeAll": false,
      "multi": false,
      "name": "deployment",
      "options": [
        {
          "selected": true,
          "text": "ion-staging",
          "value": "ion-staging"
        }
      ],
      "query": "ion-staging",
      "refresh": 0,
      "type": "custom"
    },
    {
      "allFormat": "glob",
      "current": {
        "text": "ion-staging-app",
        "value": "ion-staging-app"
      },
      "datasource": null,
      "hide": 0,
      "includeAll": false,
      "multi": false,
      "name": "app_elb_name",
      "options": [
        {
          "selected": true,
          "text": "ion-staging-app",
          "value": "ion-staging-app"
        }
      ],
      "query": "ion-staging-app",
      "refresh": 0,
      "type": "custom"
    },
    {
      "allFormat": "glob",
      "current": {
        "text": "staging",
        "value": "staging"
      },
      "datasource": null,
      "hide": 0,
      "includeAll": false,
      "multi": false,
      "name": "deployment_short",
      "options": [
        {
          "selected": true,
          "text": "staging",
          "value": "staging"
        }
      ],
      "query": "staging",
      "refresh": 0,
      "type": "custom"
    }
  ]
}
```

Diff of JSON before and after import
```
─➤  diff /ops/dash-sorted.json /ops/dash-after-import-sorted.json
7c7
<   "id": null,
---
>   "id": 6,
1340c1340
<         "hide": 2,
---
>         "hide": 0,
1362c1362
<         "hide": 2,
---
>         "hide": 0,
1384c1384
<         "hide": 2,
---
>         "hide": 0,
1432c1432
<   "version": 4
---
>   "version": 1
```

**Comment by bergquist** [comment]
Seems like a bug. 

But perhaps you could use constant template variables to solve this?

**Comment by octalthorpe** [comment]
Constant template?  I like the sounds of that, where is that documented? I cannot seem to find anything about it.

**Comment by octalthorpe** [comment]
Constant type works, exactly what I needed.  Thanks!.
I can leave this open if one wishes to fix the Import hidden bug.

**Comment by torkelo** [comment]
seems to fixed in 3.1, unable to replicate this on http://play.grafana.org/
